### ISSUE TITLE
Bigger models

### ISSUE THREAD

**Issue by avostryakov** [Helsinki-NLP/Tatoeba-Challenge#15]
Hi!

Did you have plans to train bigger models for the main Europinian language pairs, like fra<>eng, spa<>eng, por<>eng, deu<>eng, etc? Now we have so a lot of training data for main pairs that it has meaning to train large transformer models to increase translation quality.

**Comment by jorgtied** [comment]
good idea - I had this on my todo list already. I will start some pilot runs with bigger transformers. Initially, there are some issues with warmup and convergence. Are there good best-practices for training larger NMT models from scratch? I have mostly done the 6+6 layer variants ...

**Comment by avostryakov** [comment]
Exiting! Thanks!

> Are there good best-practices for training larger NMT models from scratch?

I didn't hear problems with training large MT models. Maybe only starting learning rate should be smaller. Facebook team trained large models (transformer.wmt19.en-de, transformer.wmt19.de-en, etc): https://github.com/pytorch/fairseq/tree/main/examples/translation. I believe it was trained from scratch.

**Comment by avostryakov** [comment]
Maybe it can be useful, hyper-parameters from facebook's research team:

```
encoder_embed_dim = getattr(args, 'encoder_embed_dim', 1024)
encoder_ffn_embed_dim = getattr(args, 'encoder_ffn_embed_dim', 8192)
encoder_attention_heads = getattr(args, 'encoder_attention_heads', 16)
decoder_embed_dim = getattr(args, 'decoder_embed_dim', 1024)
decoder_ffn_embed_dim = getattr(args, 'decoder_ffn_embed_dim', 8192)
decoder_attention_heads = getattr(args, 'decoder_attention_heads', 16)
dropout = getattr(args, 'dropout', 0.3)
attention_dropout = getattr(args, 'attention_dropout', 0.1)

lr = 5e-4 
lr-scheduler = inverse_sqrt 
warmup-updates = 4000
weight-decay = 0.0001
```

It's strange but it looks like they use 6 layers for encoder and decoder

**Comment by avostryakov** [comment]
Hi, @jorgtied! Any news about larger models?

**Comment by jorgtied** [comment]
Thanks for the reminder. Progress is a bit slower than expected. There are actually some bigger models already but I need to improve how to mark them to make it more visible. So far, only for a selected number of languages that I wanted to test first. It all takes longer and more care than I anticipated in the beginning. There are models for fin-swe and eng-fin for example. https://github.com/Helsinki-NLP/Tatoeba-Challenge/tree/master/models/eng-fin
